typescript.html.markdown

```diff
@@ -14,100 +14,111 @@ This article will focus only on TypeScript extra syntax, as oposed to [JavaScrip
+// There are 3 basic types in TypeScript
 var isDone: boolean = false;
 var lines: number = 42;
 var name: string = "Anders";
 
-//..When it's impossible to know, there is the "Any" type
+// When it's impossible to know, there is the "Any" type
 var notSure: any = 4;
 notSure = "maybe a string instead";
 notSure = false; // okay, definitely a boolean
 
-//For collections, there are typed arrays and generic arrays
+// For collections, there are typed arrays and generic arrays
 var list: number[] = [1, 2, 3];
-//Alternatively, using the generic array type
+// Alternatively, using the generic array type
 var list: Array<number> = [1, 2, 3];
 
-//For enumerations:
+// For enumerations:
 enum Color {Red, Green, Blue};
 var c: Color = Color.Green;
 
-//Lastly, "void" is used in the special case of a function not returning anything
+// Lastly, "void" is used in the special case of a function returning nothing
 function bigHorribleAlert(): void {
   alert("I'm a little annoying box!");
 }
 
-//Functions are first class citizens, support the lambda "fat arrow" syntax and use type inference
-//All examples are equivalent, the same signature will be infered by the compiler, and same JavaScript will be emitted
-var f1 = function(i: number) : number { return i * i; }
-var f2 = function(i: number) { return i * i; } //Return type infered
-var f3 = (i : number) : number => { return i * i; }
-var f4 = (i: number) => { return i * i; } //Return type infered
-var f5 = (i: number) =>  i * i; //Return type infered, one-liner means no return keyword needed
+// Functions are first class citizens, support the lambda "fat arrow" syntax and
+// use type inference
 
-//Interfaces are structural, anything that has the properties is compliant with the interface
+// The following are equivalent, the same signature will be infered by the
+// compiler, and same JavaScript will be emitted
+var f1 = function(i: number) : number { return i * i; }
+// Return type inferred
+var f2 = function(i: number) { return i * i; }
+var f3 = (i: number) : number => { return i * i; }
+// Return type inferred
+var f4 = (i: number) => { return i * i; }
+// Return type inferred, one-liner means no return keyword needed
+var f5 = (i: number) =>  i * i; 
+
+// Interfaces are structural, anything that has the properties is compliant with
+// the interface
 interface Person {
   name: string;
-  //Optional properties, marked with a "?"
+  // Optional properties, marked with a "?"
   age?: number;
-  //And of course functions
+  // And of course functions
   move(): void;
 }
 
-//..Object that implements the "Person" interface
-var p : Person = { name: "Bobby", move : () => {} }; //Can be treated as a Person since it has the name and age properties
-//..Objects that have the optional property:
+// Object that implements the "Person" interface
+// Can be treated as a Person since it has the name and move properties
+var p : Person = { name: "Bobby", move : () => {} };
```

Comment:
Is it customary to have a space between the variable and the colon? The rest of this page seems to not have spaces.